From 6e063e6d879af29122cee7b5b10d8add62727947 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Mon, 30 Nov 2015 10:55:12 +0800
Subject: [PATCH 0710/5242] MLK-11918-2 video: mxc ipuv3 fb: Return err in
 ->fb_check_var if fbi_tmp is NULL

commit  d2956c23ee68eedf55c68e96fe8f37c08eb9adb6 from
https://source.codeaurora.org/external/imx/linux-imx.git

It has to be successful to find a relevant background framebuffer for a
foreground framebuffer.  So, we should return error code if it happens
to fail.

This patch fixes the following issue reported by Coverity:
Dereference after null check (FORWARD_NULL)
var_deref_op: Dereferencing null pointer fbi_tmp.
		if (fbi_tmp->var.vmode & FB_VMODE_INTERLACED)
			var->vmode |= FB_VMODE_INTERLACED;
		else
			var->vmode &= ~FB_VMODE_INTERLACED;

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mxc_ipuv3_fb.c |   10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c b/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c
index 9f71693..d3f4825 100644
--- a/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c
+++ b/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c
@@ -1566,11 +1566,15 @@ static int mxcfb_check_var(struct fb_var_screeninfo *var, struct fb_info *info)
 		bg_yres = var->yres;
 
 		fbi_tmp = found_registered_fb(MEM_BG_SYNC, mxc_fbi->ipu_id);
-		if (fbi_tmp) {
-			bg_xres = fbi_tmp->var.xres;
-			bg_yres = fbi_tmp->var.yres;
+		if (!fbi_tmp) {
+			dev_err(info->device,
+				"cannot find background fb for overlay fb\n");
+			return -EINVAL;
 		}
 
+		bg_xres = fbi_tmp->var.xres;
+		bg_yres = fbi_tmp->var.yres;
+
 		ipu_disp_get_window_pos(mxc_fbi->ipu, mxc_fbi->ipu_ch, &pos_x, &pos_y);
 
 		if ((var->xres + pos_x) > bg_xres)
-- 
1.7.9.5

